test/common/http/conn_manager_impl_test.cc

```diff
@@ -57,6 +57,10 @@ class HttpConnectionManagerImplTest : public Test, public ConnectionManagerConfi
   struct RouteConfigProvider : public Router::RouteConfigProvider {
     // Router::RouteConfigProvider
     Router::ConfigConstSharedPtr config() override { return route_config_; }
+    const std::string& versionInfo() const override {
+      static const std::string* version = new std::string("");
```

Comment:
This pattern happens a lot (and in the Envoy code base in general). I wonder if there's some way to make this less boiler platey.

Reply:
+1, I would add a small macro in singleton somewhere like `SINGLETON_STRING` or something. If you search for `new std::string` you could probably find most of the instances quickly.